### PR TITLE
Test(GUI): Adding ui tests and CI integration.

### LINKED ISSUE #166: Setup UI-Tests (Library 3)
## User story
1. As a user
2. I want that the user interface is tested
3. So that I can make sure everything works

## Acceptance criteria
* a framework for ui tests was selected
* one example UI-Tests is implemented in this framework
* the example test passes green
* the UI-Tests are integrated into the CI-Pipeline

## Definition of done (DoD)
* Create feature branch
* Implement the functionality
* Opened pull request and assigned reviewer
* If necessary: required changes are implemented
* Delete feature branch
* If necessary: Component Test has been written
* Code review has been completed and code has been merged

## DoD general criteria
* Feature has been fully implemented
* Feature has been merged into the mainline
* All acceptance criteria were met
* Product owner approved features
* All tests are passing
* Developers agreed to release

### PR DESCRIPTION
**Pull request by simonsasse**
closes #166

### CODE REVIEW

**Comment by simonsasse** [comment]
SO the licenseCheck fails due to junit using EPL1.0 and hamcrest using 2-clause BSD.
BSD is no problem, its not copyleft
But EPL1.0 is copyleft.  [info](https://mvnrepository.com/artifact/junit/junit/4.13.2)
Problem: The UI testing framework needs this old jUnit4 version..

We should ask eSolution if we can use EPL license for our testing framework.

**Comment by simonsasse** [comment]
We can use the EPL1.0 license for testing purposes. So I excluded the epl1.0 license but only for JUnit, so we don't accidentally include a production library using epl1.0

File: GUI/build.gradle.kts [hunk 1]
```diff
@@ -36,8 +36,6 @@ dependencies {
     // for ui tests
     @OptIn(org.jetbrains.compose.ExperimentalComposeLibrary::class)
     implementation(compose("org.jetbrains.compose.ui:ui-test-junit4"))
```
Comment (akriese) [inline GUI/build.gradle.kts:38]: Won't this be included in the actual compiled binaries/jars if `implementation` is used instead of `testImplementation`?

**Comment by akriese** [comment]
Please add some documentation to the local README.md and possibly wiki.
Also, is there a way to execute the GUI test without triggering the DifferenceGenerator tests? `./gradlew test` includes those too at least

**Comment by simonsasse** [comment]
Entry in wiki is also added.

**Review by akriese — APPROVED** [review]

**Pull request merged**